onnxruntime/python/tools/transformers/optimizer.py

```diff
@@ -174,7 +183,15 @@ def optimize_by_onnxruntime(
         else:
             providers.append("CUDAExecutionProvider")
 
-    onnxruntime.InferenceSession(onnx_model_path, sess_options, providers=providers, **kwargs)
+    # For large model, extract external data from model and add to session options
+    if isinstance(onnx_model, ModelProto):
+        if save_as_external_data:
```

Comment:
This function is a public API.
I do not think this flag means the onnx_model object has external data not loaded into memory.
It just mean that user want to save external data to another file.

Reply:
Makes sense. Will update this.

Reply:
We can wrap the follow into a helper function:

https://github.com/microsoft/onnxruntime/blob/06fe4f31131a6873a295ba47ed60f4cb16584296/onnxruntime/python/tools/transformers/optimizer.py#L320-L323

Reply:
Yeah I'm also doing this. I'm going to move this to utils so it can be used in both places.